src/com/linkedin/parseq/internal/Continuations.java

```diff
@@ -1,7 +1,6 @@
 package com.linkedin.parseq.internal;
 
-import java.util.ArrayDeque;
-import java.util.Deque;
+import java.util.Stack;
```

Comment:
Stack is not optimal data structure here because it is unnecessarily synchronized.